### PR TITLE
feat(frontend/shared): header component bb-91

### PR DESCRIPTION
**Pull request by AndriiLevin**
**Summary**
This PR implements the Header component for the application.

**Changes**

Model: Updated the `UserDto` model to include a new avatar field.
Component: Created the `Header` component, which displays the user's avatar and name.
Subcomponent: Added the` UserHeaderInfo` component as part of the Header implementation.
Styles: 

- Added CSS styles for the `Header`/` UserHeaderInfo`  component to ensure proper layout and design consistency.
- Added the Nunito font to the project and applied it to the Header component for consistent typography.

**Additional Notes**

The `Header` component does not include the chat icon and dropdown icon, as per the task requirements.

**Screenshots**

Screenshot 2024-08-22 at 13 19 19

### CODE REVIEW

File: apps/frontend/src/assets/css/fonts/nunito.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@font-face {
+	font-family: Nunito;
+	font-style: normal;
+	font-weight: 600;
+	src: url("./Nunito-SemiBold.ttf") format("truetype");
+}
```
Comment (fshabanov) [inline apps/frontend/src/assets/css/fonts/nunito.css:6]: Font will be applied globally in another ticket, remove this

File: apps/frontend/src/assets/img/default-avatar.png
Comment (fshabanov) [inline apps/frontend/src/assets/img/default-avatar.png:1]: The default avatar should be something like this
image

File: apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import defaultAvatar from "~/assets/img/default-avatar.png";
+import { type UserDto } from "~/libs/types/types.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	user: UserDto;
+};
+
+const UserHeaderInfo: React.FC<Properties> = ({ user }) => {
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx:12]:
```suggestion
const UserHeaderInformation: React.FC<Properties> = ({ user }) => {
```
B16

File: apps/frontend/src/libs/components/header/header.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+import { useAppSelector } from "~/libs/hooks/use-app-selector/use-app-selector.hook.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/header.tsx:3]:
```suggestion
import { useAppSelector } from "~/libs/hooks/hooks.js";
```

File: apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx:1]:
```suggestion
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx (hunk 4 above)
Comment (AndriiLevin) [reply apps/frontend/src/libs/components/header/components/user-header-info/user-header-info.tsx]: After removing the React import, I'm encountering an issue with prop passing, which is leading to errors.
`ESLint: 'user' is missing in props validation(react/ prop-types)`

 Although the following workaround fixes it, it's likely not the correct approach.

`const UserHeaderInformation: React.FC<Properties> = ({ user }: Properties) => {`

And also half of your components have import React.

**Comment by AndriiLevin** [comment]
Screenshot 2024-08-23 at 15 03 24

**Review by fshabanov — APPROVED** [review]

**Pull request merged**